src/sentry/taskworker/registry.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import annotations
+
+import logging
+import time
+from collections.abc import Mapping
+from datetime import timedelta
+from typing import Any
+from uuid import uuid4
+
+import orjson
+from arroyo.backends.kafka import KafkaPayload, KafkaProducer
+from arroyo.types import Topic as ArroyoTopic
+from google.protobuf.timestamp_pb2 import Timestamp
+from sentry_protos.sentry.v1.taskworker_pb2 import RetryState, TaskActivation
+
+from sentry.conf.types.kafka_definition import Topic
+from sentry.taskworker.retry import FALLBACK_RETRY, Retry
+from sentry.taskworker.task import Task
+from sentry.utils.kafka_config import get_kafka_producer_cluster_options, get_topic_definition
+
+logger = logging.getLogger(__name__)
+
+
+class TaskNamespace:
+    """
+    Task namespaces link topics, config and default retry mechanics together
+    All tasks within a namespace are stored in the same topic and run by shared
+    worker pool.
+    """
+
+    __registered_tasks: dict[str, Task]
+    __producer: KafkaProducer | None = None
+
+    def __init__(self, name: str, topic: str, deadletter_topic: str, retry: Retry | None):
+        # TODO(taskworker) implement default deadlines for tasks
+        self.name = name
+        self.topic = topic
+        self.deadletter_topic = deadletter_topic
+        self.default_retry = retry
+        self.__registered_tasks = {}
+
+    @property
+    def producer(self) -> KafkaProducer:
+        if self.__producer:
+            return self.__producer
+        cluster_name = get_topic_definition(Topic.TASK_WORKER)["cluster"]
+        producer_config = get_kafka_producer_cluster_options(cluster_name)
+        self.__producer = KafkaProducer(producer_config)
+
+        return self.__producer
+
+    @producer.setter
+    def producer(self, producer: KafkaProducer) -> None:
+        self.__producer = producer
+
+    @producer.deleter
+    def producer(self) -> None:
+        self.__producer = None
+
+    def get(self, name: str) -> Task:
+        if name not in self.__registered_tasks:
+            raise KeyError(f"No task registered with the name {name}. Check your imports")
+        return self.__registered_tasks[name]
+
+    def contains(self, name: str) -> bool:
+        return name in self.__registered_tasks
+
+    def register(
+        self,
+        name: str,
+        idempotent: bool | None = None,
+        deadline: timedelta | int | None = None,
+        retry: Retry | None = None,
+    ):
```

Comment:
since this is brand new infra -- please add this file to the "stronger" types list in `pyproject.toml` (and fill out the rest of the types here

as it's written this is discarding type information and makes every user of this registry function unsound